packages/steiger/src/models/config/build-validation-scheme.spec.ts

```diff
@@ -0,0 +1,311 @@
+      {
+        rules: {
+          rule1: ['warn', {}],
+          rule2: ['error', {}],
+        },
+      },
+    ] as Config
+    const scheme = buildValidationScheme(config)
+
+    expect(scheme.parse(config)).toEqual(checkToAnyFunction(config))
+  })
+
+  it('should successfully validate a config with several objects', () => {
+    const config = [
+      dummyPlugin,
+      {
+        rules: {
+          rule1: 'off',
+        },
+      },
+      {
+        rules: {
+          rule2: 'warn',
+        },
+      },
+    ] as Config
+
+    const scheme = buildValidationScheme(config)
+
+    expect(scheme.parse(config)).toEqual(checkToAnyFunction(config))
+  })
+
+  it('should successfully validate a config with ignores', () => {
+    const config = [
+      dummyPlugin,
+      {
+        ignores: ['/shared'],
+        rules: {
+          rule1: 'off',
+        },
+      },
+      {
+        ignores: ['/shared'],
+        rules: {
+          rule2: 'warn',
+        },
+      },
+    ] as Config
+    const scheme = buildValidationScheme(config)
+
+    expect(scheme.parse(config)).toEqual(checkToAnyFunction(config))
+  })
+
+  it('should successfully validate a config with files', () => {
+    const config = [
+      dummyPlugin,
+
+      {
+        files: ['/shared'],
+        rules: {
+          rule1: 'off',
+        },
+      },
+      {
+        files: ['/shared'],
+        rules: {
+          rule2: 'warn',
+        },
+      },
+    ] as Config
+    const scheme = buildValidationScheme(config)
+
+    expect(scheme.parse(config)).toEqual(checkToAnyFunction(config))
+  })
+
+  it('should successfully validate a config with files and ignores', () => {
+    const config = [
+      dummyPlugin,
+      {
+        files: ['/shared'],
+        ignores: ['**/*.test.ts'],
+        rules: {
+          rule1: 'off',
+        },
+      },
+      {
+        files: ['/shared'],
+        ignores: ['**/*.test.ts'],
+        rules: {
+          rule2: 'warn',
+        },
+      },
+    ] as Config
+    const scheme = buildValidationScheme(config)
+
+    expect(scheme.parse(config)).toEqual(checkToAnyFunction(config))
+  })
+
+  it('should throw an error if no plugins are provided', () => {
+    const config = [
+      {
+        rules: {
+          rule1: 'off',
+        },
+      },
+    ] as Config
+    expect(() => buildValidationScheme(config)).toThrow('At least one rule must be provided by plugins!')
+  })
+
+  it('should throw an error if no config objects are provided', () => {
+    const config = [dummyPlugin] as Config
+    const scheme = buildValidationScheme(config)
+
+    expect(() => scheme.parse(config)).toThrow('At least one config object must be provided!')
+  })
+
+  it('should throw an error if a config object without rules is provided', () => {
+    const config = [dummyPlugin, {}] as Config
+    const config1 = [dummyPlugin, { files: ['/shared'] }] as Config
+    const config2 = [dummyPlugin, { ignores: ['**/*.test.ts'] }] as Config
```

Comment:
In this case `dummyPlugin` contains only the plugin object itself without any config objects, so:
1. `{}` is not allowed because only 3 shapes of objects are allowed in a config (Plugin, ConfigObject, GlobalIgnore)
2. This shape `{ files: ['/shared'] }` is not valid because it's not a plugin, global ignore, or config object.
3. `{ ignores: ['**/*.test.ts'] }`. Even though it's a valid global ignore, this config still misses a single config object, which is nonsense, as no rules are configured to be run